### ISSUE TITLE
[Windows 10][Client] Sync client cannot rename download temporary file

### ISSUE THREAD

**Issue by DaveJ61** [owncloud/client#3914]
I have the latest owncloud server on my local lan file server running Ubuntu Server 14.04. I also have a Xubuntu 15.04 machine and my Windows 10 machine. Both of these are running the latest clients 2.0.1 (build 5446). As far as I know, nothing has changed and I haven't changed any configuration.
The linux client works perfectly for me but the windows 10 client has issues. It is not syncing properly and I end up with files in the local folders that are hidden with a preceding dot and also have a 3 or 4 digit extension which I believe is a temporary file designation. The client activity is showing access is denied. This is certainly nothing to do with the server as nothing has changed on it and both my android and linux clients work without any problems.

This problem appears to be similar to https://forum.owncloud.org/viewtopic.php?f=14&t=30195 but I have the latest client, I did uninstall the old one first and this has been working and now has failed.

If I upload a file from the windows machine by dumping it into the local owncloud directory, that works fine. However, if I dump a file silently, or by way of html access, the file will get synced to my windows 10 machine, but will be hidden, with a preceding dot and an extra 3 or 4 digit extension. I have attached a sample directory from my windows folders showing the problem. The logs show that everything seems to be working and then on 1st Oct, everything started failing and cannot download new files as access is denied. My account on the windows machine has an Administrator status so this should not be happening, surely? I originally solved this problem by installing as an administrator, even though my account is one. But it has now reverted and the problem is in existence again. The linux client is working flawlessly as expected.
To complicate matters, for some reason, I can't attach files to this issue either. It just says "Something went really wrong, and we can’t process that file, please try again". I can't attach the log file either and it's a bit large to copy and paste here.

So here is a small directory displaying the problem:-

 Volume in drive C is 512SSD
 Volume Serial Number is 56D4-2F64

 Directory of C:\Users\David\ownCloud\InstantUpload\photos\Bike Rides\Harringworth

10/03/2015  09:22 AM    <DIR>          .
10/03/2015  09:22 AM    <DIR>          ..
10/01/2015  04:50 PM           493,601 .20151001_142622.jpg.~fa7
10/01/2015  04:51 PM         4,326,327 .20151001_142647.jpg.~3a0a
10/01/2015  04:51 PM         4,217,655 .20151001_142649.jpg.~458d
10/01/2015  04:51 PM         4,548,860 .20151001_152230.jpg.~76c7
10/01/2015  04:52 PM         4,642,822 .20151001_152234.jpg.~2f4f
10/01/2015  04:52 PM         4,344,777 .20151001_152239.jpg.~2db3
10/01/2015  04:52 PM         4,601,793 .20151001_152241.jpg.~5857
10/03/2015  09:22 AM                 0 dir.txt
               8 File(s)     27,175,835 bytes
               2 Dir(s)  58,557,018,112 bytes free

**Comment by Dianafg76** [comment]
I tested this issue, and In Win10 you can't see the hidden files
Desktop v ownCloud-2.0.2.5518-nightly20151006-setup.exe 
{"installed":true,"maintenance":false,"version":"8.1.3.0","versionstring":"8.1.3","edition":"Enterprise"}

**Comment by dragotin** [comment]
I do not understand correctly what you are doing, expecially *if I dump a file silently, or by way of html access, the file will get synced to my windows 10 machine, but will be hidden*. Please try to explain what you do to get the error. Also, please create a client log file as described in the client troubleshooting documentation. Thanks,

**Comment by DaveJ61** [comment]
I am just using the windows client as it should be used. I thought that what I had put in my original post was quite clear. Also, quite clear on why I cannot attach a a log file. I get a message saying "Something went really wrong, and we can’t process that file. Try again." when trying to attach the file. I just tried it again. I even renamed the file to have a txt extension as .log files are not recognised.


Again... if I add files by uploading them by html then they get get synced to my windows 10 machine but with a directory listing as previously described and as per the pasted structure above. They are hidden. Sure I could un-hide them, remove the dot from the front and remove the extra extension, and then I could still use the file as normal. But I sure am not doing that with every file or picture I upload from my phone or other pc.

**Comment by dragotin** [comment]
Sorry, not clear enough for  me. Sometimes I have a hard time  to understand this sync stuff ;-)
You mean if you upload a file during the web interface, it does not come back to the client but remains as hidden file?

These hidden files stem from the download process of the client. It downloads a file from the server into the hidden file and once that is finished, it renames it to the final name. The last step fails for you, no idea why actually... It is a bit strange. Can you create a file in that directory manually?

More we would see in the logfile. You could search for a file that fails in the logfile and paste 100 lines around that here.

**Comment by DaveJ61** [comment]
Ok. I have now installed the owncloud client onto my hard drive but outside of the "Program Files" directory. I have stuck it into an apps folder and, touch wood, everything appears to be working at the moment. I have kept the data location in my user directory as per standard installation.
I'm going to test for a few days and see what happens.

**Comment by dragotin** [comment]
yes, you need write access in the sync directory that you use.

**Comment by DaveJ61** [comment]
I do have write access to the data... it's my user folder. I can read/write and do anything, my account is an administrator account anyway. It should work regardless as a standard installation, but it does not. Anyway, I will test.

**Comment by DaveJ61** [comment]
So.... I get up this morning and turn my PC on and... owncloud client has packed up again. After working away merrily yesterday syncing files I have added, it seems overnight it has forgotten settings. The checkbox for confirming whether folders larger than 500Mb has miraculously become checked again. Also, despite receiving desktop alerts that blah number of files have been updated... several times, I see in the activity tab, that it has tried to download all the files again, despite them already being in the sync folder and furthermore, all of the files it has "downloaded" are of access denied status and they are all hidden again, thereby overwriting the originals with crap. Oh and please note the conflicting copy errors.

![owncloudactivity](https://cloud.githubusercontent.com/assets/1816367/10358447/58b748e0-6d86-11e5-8c47-a3bdfe5bf2ae.jpg)

In an effort to try and correct this problem, I deleted the server config, and in doing so the server tab has disappeared. The only way to get that back was to close down owncloud and restart it, however, the old server config was still there.
I changed the local folder to somewhere else and it starts the sync but access is denied, but when I closed the client down, it forgot my previous settings.... and resumed syncing to the old folder I had configured. What a mess...
I'm tired of trying to get this working and I really have better things to do. The linux client is working a treat, so is android. Damned microsoft isn't.
Strange I can attach pictures here but not text files.

**Comment by DetunizedGravity** [comment]
I have been experiencing the same behavior here on the client 2.0.1 (build 5446), on a Windows 7 Family Premium, for a while. Server is v8.1.3 on Debian 8 but it seems orthogonal to the issue.
The client fails to rename the temporary file with a "access refused" message in the log, when downloading it. Locally created files are uploaded correctly, and then can be renamed from the server WebUI without issue (the new name is propagated). Any remotely created file is downloaded locally but not correctly renamed to the desired name. More surprising, if such a file is remotely renamed (WebUI), the local temporary file is renamed accordingly, but stays in a temporary state.
If I manually create a local copy a of remotely created file before syncing, the remote version is downloaded as a temporary file, without consideration of the local copy. Both versions of the file coexist. Before and during sync the local copy would be flagged with the blue icon. After sync, it would be flagged as erroneous (red icon). Such files are downloaded again and again, on each sync attempt. By the way, since my cell phone is currently my only Internet access, this behavior used all my allowance for October (50 gigabytes) in 4 or 5 days. Now working on reduced bandwidth.

**Comment by guruz** [comment]
@DetunizedGravity @DaveJ61 Do you have any virus scanners or similar running that prevent the client from renaming?
Does it work if you have all applications closed that access the files (either the download temporary file or the destination file it should have been renamed to)

Note that there are also tools like: http://lockhunter.com/

@dragotin @ogoffart This is about the client's local temporary download files not being able to be renamed, see the file name pattern e.g. ```.20151001_142649.jpg.~458d```

**Comment by DaveJ61** [comment]
Hi,
I've just done a re-install a few days ago of my windows 10 and I'm currently testing. Everything seems to be working fine now. From scouring the Windows 10 forums, a lot of people are having permissions issues where a program worked flawlessly in windows 7 or 8, it stopped working after the upgrade to 10.
I think it looks good for now but if I have any more problems, it looks like it is a Microsoft problem, not Owncloud.

Regards

David

**Comment by guruz** [comment]
Thanks, I'm closing this for now.

**Comment by DetunizedGravity** [comment]
Hello and sorry for the late answer. I have had an awful lot of real life
issues to deal with lately (including a broken hand) and hence let this
topic slip under my radar.

As far as I am concerned, the problem was never related to Windows 10. It
happened on Win7. Unfortunately I will not be able to help diagnose it or
even try and replicate until at least next July, for various reasons. By
that time all my PCs should have been migrated to either Win10 or Linux, so
the issue may become moot.
Until then, I will try and keep the total volume of data sync'ed through my
cell phone connection to a minimum.

Best regards,
  Xavier


2015-11-10 16:27 GMT+01:00 Markus Goetz <notifications@github.com>:

**Comment by marcoesch** [comment]
Hello,
Why has this issue been closed? Is there a solution?
I have the Windows client on 2 Windows 10 machines same version of the client. One is running ok the other one is having exactly this issue. (Actually it happens on version 2.2.3 (B 6307) and the latest one V. 2.2.4
Please advice
Thanks,
Marc

**Comment by guruz** [comment]
@marcoesch This issue is already a year old. Please create a new one, including exact descriptions on what you are doing and which versions etc
(There is an issue template when creating a new issue)
